lang/c++/CMakeLists.txt

```diff
@@ -69,17 +70,16 @@ endif()
 
 if (CMAKE_COMPILER_IS_GNUCXX)
     set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wextra -Wduplicated-cond -Wduplicated-branches -Wlogical-op -Wuseless-cast -Wconversion -pedantic -Werror")
-if (AVRO_ADD_PROTECTOR_FLAGS)
-    set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fstack-protector-all -D_GLIBCXX_DEBUG")
-    # Unset _GLIBCXX_DEBUG for avrogencpp.cc because using Boost Program Options
-    # leads to linking errors when compiling with _GLIBCXX_DEBUG as described on
-    # https://stackoverflow.com/questions/19729036/
-    set_source_files_properties(impl/avrogencpp.cc PROPERTIES COMPILE_FLAGS "-U_GLIBCXX_DEBUG")
-endif ()
 endif ()
 
 if (AVRO_BUILD_TESTS OR AVRO_USE_BOOST)
-    find_package (Boost 1.38 REQUIRED COMPONENTS system)
+    # Boost 1.70 and above provide a BoostConfig.cmake package configuration file.
+    find_package (Boost 1.70 REQUIRED CONFIG COMPONENTS system)
+    if (TARGET Boost::system)
+        message("Found Boost version: ${Boost_VERSION}")
+    else ()
+        message(FATAL_ERROR "Boost::system not found")
+    endif ()
```

Comment:
You mark this dependency as required. I think this check if target exists is not needed. CMake will fail if it does not find Boost and all requested components.

Reply:
I just want to make sure that the target with `Boost::` namespace exists. Same for `Snappy::snappy` and `ZLIB::ZLIB`

Reply:
Snappy and ZLIB are not marked as required.

Reply:
Boost is not required. It is required only when `AVRO_BUILD_TESTS` or `AVRO_USE_BOOST` is enabled.

Reply:
I think what @mitjap was trying to say is that this `if` statement is superfluous as `find_package` is called with a `REQUIRED` keyword. It will fail by itself when `Boost::system` is not found, no need for doing it explicitly

Reply:
This is slightly different. CMIW, sometimes `Boost::system` target is not available even if `Boost_FOUND` is true since older Boost distributions may not add `Boost::` namespace.

Reply:
I have changed the message. Please check again.